### PR TITLE
feat: add Dockerfile

### PR DESCRIPTION
**Pull request by smoya**
**Description**

This PR adds a `Dockerfile` to this project so developers can run all test suites without configuring any other dependency locally. Specially useful for blackbox test suite. 
In fact, you can run any other command you want. Once we have a CLI, we could publish the Docker image as well so people can start using it directly from a Docker registry.

Docs has been updated.

### CODE REVIEW

**Review by jonaslagoni — COMMENTED** [review]
@smoya I think we should make sure the docker image actually works with the CI, maybe we can adapt the workflow from the generator to build and test it? 🤔 - https://github.com/asyncapi/generator/blob/master/.github/workflows/release-docker.yml

**Comment by smoya** [comment]
> @smoya I think we should make sure the docker image actually works with the CI, maybe we can adapt the workflow from the generator to build and test it? 🤔 - https://github.com/asyncapi/generator/blob/master/.github/workflows/release-docker.yml

Good point. However, the main test suite runs on a shared workflow: https://github.com/asyncapi/modelina/blob/master/.github/workflows/if-nodejs-pr-testing.yml (comes from our .github repository).

What about making https://github.com/asyncapi/modelina/blob/master/.github/workflows/blackbox-testing.yml use the docker image instead?

Regarding the release of the Docker image, I would not include it yet. Once we have a reason to provide public Docker images  (such as having a CLI tool) we can add that step.

**Comment by jonaslagoni** [comment]
> Regarding the release of the Docker image, I would not include it yet. Once we have a reason to provide public Docker images (such as having a CLI tool) we can add that step.

Yea that's my point, remove the release step from the workflow so it only builds the docker image and runs the tests afterward 👍 

> What about making https://github.com/asyncapi/modelina/blob/master/.github/workflows/blackbox-testing.yml use the docker image instead?

Good suggestion! Lets try that, so it builds the docker image and then runs the tests inside the docker image 👍

**Comment by smoya** [comment]
> Good suggestion! Lets try that, so it builds the docker image and then runs the tests inside the docker image

I'm gonna add this in a follow up issue as it is not that trivial as it sounds. We would need to reuse docker images somehow in order to not penalize CI time, among others.

File: package.json [hunk 1]
```diff
@@ -62,9 +62,12 @@
     "build:cjs": "tsc",
     "build:esm": "tsc --project tsconfig.json --module ESNext --outDir ./lib/esm",
     "build:types": "tsc --project tsconfig.json --declaration --emitDeclarationOnly --declarationMap --outDir ./lib/types",
+    "build:docker": "docker build -q -t asyncapi/modelina .",
     "test": "cross-env CI=true jest --coverage --testPathIgnorePatterns ./test/blackbox",
     "test:blackbox": "cross-env CI=true jest ./test/blackbox",
     "test:watch": "jest --watch",
+    "test:docker": "npm run build:docker && docker run asyncapi/modelina npm run test",
```
Comment (jonaslagoni) [inline package.json:69]: Would `docker:test` not make more sense? As we are testing using docker, we don't test docker 🤔

File: package.json [hunk 2]
```diff
@@ -62,9 +62,12 @@
...
+    "test:docker:blackbox": "npm run build:docker && docker run asyncapi/modelina npm run test:blackbox",
```
Comment (jonaslagoni) [inline package.json:70]: Same here as above

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Does the script naming make sense? 🤔

File: package.json (hunk 1 above)
Comment (jonaslagoni) [reply package.json:69]: hmm... I guess it might be better to stay as is, in terms of how the rest of the scripts are 🤔

Comment (jonaslagoni) [reply package.json:69]: Thoughts?

Comment (smoya) [reply package.json:69]: Yeah, I'm happy with both naming patterns. At the end you could also read this as `test:*` is all related to testing. But Agree prefixing with `docker:` instead could make more sense. Changing!

Comment (smoya) [reply package.json]: Done!

File: package.json (hunk 2 above)
Comment (smoya) [reply package.json]: Fixed

**Review by jonaslagoni — APPROVED** [review]
LGTM 👍

**Comment by smoya** [comment]
Test coverage failed because of a random weird error. I’m gonna skip it as I did not add any test or change it (no work on code)

**Comment by derberg** [comment]
yo, just a comment that this probably should not be a `feat` triggering release as release artifact did not change, and you are not publishing docker image to dockerhub

**Comment by smoya** [comment]
> yo, just a comment that this probably should not be a `feat` triggering release as release artifact did not change, and you are not publishing docker image to dockerhub

Good catch. My fault, now it's a bit late unfortunately. Will take care next time.

**Comment by jonaslagoni** [comment]
Yea agreed you are right @derberg, will keep it in mind 👍

**Comment by derberg** [comment]
that is fine folks, I left my comment as a future remark to always keep in mind to look at the title, especially if one day we find time to automate merge process by the bot

**Pull request merged**